cache.go

```diff
@@ -80,7 +80,12 @@ func (c *cache) Update(req *pb.FindFullHashesRequest, resp *pb.FindFullHashesRes
 		if c.pttls[fullHash] == nil {
 			c.pttls[fullHash] = make(map[ThreatDescriptor]time.Time)
 		}
-		dur := time.Duration(tm.GetCacheDuration().Seconds) * time.Second
+		var dur time.Duration
+		if tmCacheDur := tm.GetCacheDuration(); tmCacheDur != nil {
+			dur = time.Duration(tm.GetCacheDuration().Seconds) * time.Second
+		} else {
+			dur = time.Second
```

Comment:
@colonelxc @dsnet Changed to use zero in https://github.com/google/safebrowsing/pull/45/commits/0f9e98561a5096bba7a8af218f8b4d929e8770f7 - thanks for the input